components/eventsrv/src/lib.rs

```diff
@@ -12,13 +12,125 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// NOTE: The sole purpose of this client is testing and debugging. It's not
-// required for normal operation.
-
 #[macro_use]
 extern crate log;
 extern crate protobuf;
 extern crate time;
 extern crate zmq;
 
 mod message;
+
+use message::event::EventEnvelope;
+use protobuf::parse_from_bytes;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use zmq::{Context, PULL, XPUB};
+
+/// Proxies messages coming into `frontend_port` out through
+/// `backend_port`, caching recent messages for new subscribers.
+///
+/// Event publishers should connect a ZMQ `PUSH` socket to
+/// `frontend_port` and send `EventEnvelope` protobuf messages to
+/// it. Publishers can connect to multiple such proxy processes;
+/// messages will be fairly-dealt to all connected proxies.
+///
+/// Event subscribers should connect a ZMQ `SUB` socket to
+/// `backend_port` to receive `EventEnvelope` protobuf
+/// messages. Subscribers should connect to all available proxy
+/// processes to ensure they receive all event messages.
+///
+/// Upon connection, subscribers will receive, from each connected
+/// proxy, the most recent messages from each service and from each
+/// ring member. Subscribers are responsible for sorting the messages
+/// received by timestamp.
+pub fn proxy(frontend_port: i32, backend_port: i32) {
+    let ctx = Context::new();
+
+    let pull_sock = ctx.socket(PULL).unwrap();
+    let pull_bind = format!("tcp://*:{}", frontend_port);
+    assert!(pull_sock.bind(&pull_bind).is_ok());
+
+    let xpub_sock = ctx.socket(XPUB).unwrap();
+    let xpub_bind = format!("tcp://*:{}", backend_port);
+    assert!(xpub_sock.bind(&xpub_bind).is_ok());
```

Comment:
These changed from `connect` to `bind` because this is the server, which is the "stable" part of the system.  Is that right?